cmd/up.go

```diff
@@ -80,9 +86,107 @@ func newUpCmd() *cobra.Command {
 				Debug:                debug,
 			}
 
-			s, err := requireStack(stack, true, opts.Display)
-			if err != nil {
-				return err
+			var c config.Map
+			var s backend.Stack
+			hasStack := false
+
+			if len(args) > 0 {
+				url := args[0]
+
+				if !workspace.IsTemplateURL(url) {
+					return errors.Errorf("%s is not a valid URL", url)
+				}
+
+				// Retrieve the template repo.
+				var repo workspace.TemplateRepository
+				if repo, err = workspace.RetrieveTemplates(url, false); err != nil {
+					return err
+				}
+
+				// List the templates from the repo.
+				var templates []workspace.Template
+				if templates, err = repo.Templates(); err != nil {
+					return err
+				}
+
+				// Make sure only a single template is found.
+				// Alternatively, we could consider prompting to choose one instead of failing.
+				if len(templates) != 1 {
+					return errors.Errorf("more than one app template found at %s", url)
+				}
+				template := templates[0]
+
+				// Create temp directory for the "virtual workspace".
+				var temp string
+				if temp, err = ioutil.TempDir("", "pulumi-up-"); err != nil {
+					return err
+				}
+
+				// TODO don't use template name/description for project name/description.
+				// Consider prompting if they are ${PROJECT} and ${DESCRIPTION}.
+				if err = template.CopyTemplateFiles(temp, true, template.Name, template.Description); err != nil {
+					return err
+				}
+
+				// Delete the template repo.
+				if err = repo.Delete(); err != nil {
+					return err
+				}
+
+				// Change the working directory to the "virtual workspace".
+				if err = os.Chdir(temp); err != nil {
+					return errors.Wrap(err, "changing the working directory")
+				}
+
+				// If it's an existing stack, use it to fetch the latest config.
+				var stackConfig config.Map
+				if stack != "" {
+					s, err = getStack(stack)
+					// TODO: if not found, offer to create it.
```

Comment:
Need to fix this up. Here we get the stack (so we can retrieve the latest config for the stack, if it exists), but return error if the stack doesn't exist. Instead we should offer to create it. And rationalize with the code below that calls `requireStack`.